static/app/utils/withCustomerDomain.tsx

```diff
@@ -0,0 +1,116 @@
+  WrappedComponent: RouteComponent,
+  options?: WithCustomerDomainOptions
+) {
+  return function WithCustomerDomainWrapper(props: P) {
+    const {customerDomain, links} = window.__initialData;
+    const {sentryUrl} = links;
+    const currentOrganization = useContext(OrganizationContext);
+
+    if (customerDomain) {
+      // Customer domain is being used on a route that has an :orgId parameter.
+      const redirectPath = `${window.location.pathname}${window.location.search}${window.location.hash}`;
+      const redirectURL = `${trimEnd(sentryUrl, '/')}/${trimStart(redirectPath, '/')}`;
+
+      if (currentOrganization) {
+        if (
+          currentOrganization.slug !== customerDomain.subdomain ||
+          !currentOrganization.features.includes('customer-domains')
+        ) {
+          window.location.replace(redirectURL);
+          return null;
+        }
+      }
+
+      const {params, routes} = props;
+
+      // Regenerate the full route with the :orgId parameter omitted.
+      const newParams = {...params};
+      Object.keys(params).forEach(param => {
+        newParams[param] = `:${param}`;
+      });
+      const fullRoute = recreateRoute('', {routes, params: newParams});
+      let orglessSlugRoute = fullRoute
+        .replace(/organizations\/:orgId\/?/, '')
+        .replace(/:orgId\/?/, '');
+
+      if (orglessSlugRoute === fullRoute) {
+        // :orgId is not present in the route, so we do not need to perform a redirect here.
+        return <WrappedComponent {...props} />;
+      }
+
+      if (options && options.redirect) {
+        const result = options.redirect.find(needle => needle.from === fullRoute);
+        if (result) {
+          orglessSlugRoute = result.to;
+        }
+      }
+
+      const orglessRedirectPath = formatPattern(orglessSlugRoute, params);
+      const redirectOrgURL = `/${trimStart(orglessRedirectPath, '/')}${
+        window.location.search
+      }${window.location.hash}`;
+
+      const paramOrgId = (params as any).orgId ?? '';
+
+      Sentry.withScope(function (scope) {
+        const wrongOrgId = paramOrgId !== customerDomain.subdomain ? 'yes' : 'no';
+        scope.setTag('isCustomerDomain', 'yes');
+        scope.setTag('customerDomain.organizationUrl', customerDomain.organizationUrl);
+        scope.setTag('customerDomain.sentryUrl', customerDomain.sentryUrl);
+        scope.setTag('customerDomain.subdomain', customerDomain.subdomain);
+        scope.setTag('customerDomain.fromRoute', fullRoute);
+        scope.setTag('customerDomain.redirectRoute', orglessSlugRoute);
+        scope.setTag('customerDomain.wrongOrgId', wrongOrgId);
+        scope.setTag('customerDomain.paramOrgId', paramOrgId);
+        scope.setContext('customerDomain', {
+          customerDomain,
+          fullRoute,
+          orglessSlugRoute,
+          redirectOrgURL,
+          routeParams: params,
+        });
+        Sentry.captureMessage('Redirect with :orgId param on customer domain');
```

Comment:
Should we set the level on this to info/warn?

This also looks like it could be high throughput once we start having customers use domains. Should we randomly sample this?

Reply:
> Should we set the level on this to info/warn?

`Sentry.captureMessage()` already sets the level to info:

<img width="695" alt="Screen Shot 2022-11-10 at 5 05 18 PM" src="https://user-images.githubusercontent.com/139499/201215877-6cffd665-5bd3-4b42-8aa9-65a85bf3491a.png">


> This also looks like it could be high throughput once we start having customers use domains. Should we randomly sample this?

I agree and I plan to add this later on.

I prefer not to randomly sample for the internal testing we will do to find unpatched links.